src/pages/auth-callback.tsx

```diff
@@ -104,8 +104,11 @@ const AuthCallbackPage: NextPageWithLayout = () => {
                 );
 
                 setStatusMessage('Redirecting to app...');
-
-                window.location.href = '/';
+                if (redirect) {
+                  window.location.href = redirect;
+                } else {
+                  window.location.href = "/";
+                }
```

Comment:
Let's change this to use Next's `useRouter()` `router.push()` or `replace()` method.

Reply:
I have tested with both `router.push` and `router.replace` but both doesn't seem to work. 
(It does redirect after create account success, but it failed to execute some logic that does not update user profile)
![image](https://github.com/near/near-discovery/assets/6027014/79817e94-841f-4ee6-963b-0752d168a0ca)

After reviewing the code, it looks like some of init procedures are only executed on page load and that's why it only works on `window.location.href`. 

It looks like much work is needed to replace this, eg (on `VmInitializer` component, we either need to update `useAccount` to add logic to refresh or get `accountId` from other source which at this point I'm not sure if this is what we want to do)

Let me know your thoughts on this, @calebjacob

Reply:
Thanks for looking into that @hcho112 - let's roll with this as it is for now. Let's maybe add a code comment next to those lines explaining why we need to use `window.location` instead of Next's router.